universe/Pathfinder.cpp

```diff
@@ -889,6 +947,34 @@ std::pair<std::list<int>, double> Pathfinder::PathfinderImpl::ShortestPath(int s
     }
 }
 
+std::pair<std::list<int>, double> Pathfinder::ShortestNonHostilePath(int system1_id, int system2_id,
+                                                                     int empire_id) const
+{ return pimpl->ShortestNonHostilePath(system1_id, system2_id, empire_id); }
+
+std::pair<std::list<int>, double> Pathfinder::PathfinderImpl::ShortestNonHostilePath(int system1_id, int system2_id,
+                                                                                     int empire_id) const
+{
+    if (empire_id == ALL_EMPIRES) {
+        ErrorLogger() << "Invalid empire " << empire_id;
```

Comment:
Why disallow `ALL_EMPIRES` here?  Why not allow weak monsters etc the ability to try and avoid empire fleets?

See also the comment in `System::CointainsHostileFleet()` code.

Reply:
Although it's an interesting idea, perhaps worth opening a design discussion for, it is certainly consistent with our blockades approach (monsters are very poor at enforcing or evading blockades) that they would likewise be poor (or even worse, if that were possible) at avoiding enemies not even in the same system as them.

Reply:
My initial impression:
When `ALL_EMPIRES` is in the context of Unowned empire, may need a special graph map for Unowned (Unowned ships possibly share visibility).
When `ALL_EMPIRES` refers to multiple empires, there is no clear determination for visibility or hostility.
I will review this and see if an alternative seems feasible.